godal.cpp

```diff
@@ -1767,4 +1767,77 @@ GDALDatasetH godalNearblack(cctx *ctx, const char *pszDest, GDALDatasetH hDstDS,
 
 	godalUnwrap();
 	return ret;
+}
+
+OGRSpatialReferenceH godalGetGCPSpatialRef(GDALDatasetH hSrcDS) {
+	return GDALGetGCPSpatialRef(hSrcDS);
+}
+
+const GCPsAndCount godalGetGCPs(GDALDatasetH hSrcDS) {
+	return GCPsAndCount {
+		.gcpList = GDALGetGCPs(hSrcDS), 
+		.numGCPs = GDALGetGCPCount(hSrcDS),
+	};
+}
+
+const char *godalGetGCPProjection(GDALDatasetH hSrcDS) {
+	return GDALGetGCPProjection(hSrcDS);
+}
+
+void godalSetGCPs(cctx *ctx, GDALDatasetH hSrcDS, int numGCPs, goGCPList GCPList, const char *pszGCPProjection) {
+	godalWrap(ctx);
+
+	// Convert `goGCPList` -> `GDAL_GCP*`
+	GDAL_GCP *GDALGCPList = static_cast<GDAL_GCP *>(CPLCalloc(numGCPs, sizeof(GDAL_GCP)));
+    GDALInitGCPs(numGCPs, GDALGCPList);
+    for (int i = 0; i < numGCPs; i++)
+    {
+		GDALGCPList[i].pszId = CPLStrdup(GCPList.pszIds[i]); 
+		GDALGCPList[i].pszInfo = CPLStrdup(GCPList.pszInfos[i]);
+        GDALGCPList[i].dfGCPPixel = GCPList.dfGCPPixels[i];
+		GDALGCPList[i].dfGCPLine = GCPList.dfGCPLines[i];
+		GDALGCPList[i].dfGCPX = GCPList.dfGCPXs[i];
+		GDALGCPList[i].dfGCPY = GCPList.dfGCPYs[i];
+		GDALGCPList[i].dfGCPZ = GCPList.dfGCPZs[i];
+    }
```

Comment:
nitpick 1: this code can be mutualized with godalSetGCPs2, right?
nitpick 2: indentation needs fixing

Reply:
We created a new function to address the duplicate code and fixed the indentation issue here